includes/class-micropub-admin.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+// For debugging purposes this will set all Micropub posts to Draft
+if ( ! defined( 'MICROPUB_DRAFT_MODE' ) ) {
+	define( 'MICROPUB_DRAFT_MODE', '0' );
+}
+
+add_action( 'plugins_loaded', array( 'Micropub_Admin', 'init' ), 10 );
+add_action( 'admin_menu', array( 'Micropub_Admin', 'admin_menu' ) );
+
+/**
+ * Micropub Admin Class
+ */
+class Micropub_Admin {
+	/**
+	 * Initialize the admin screens.
+	 */
+	public static function init() {
+		$cls = get_called_class();
+
+		add_action( 'admin_init', array( $cls, 'admin_init' ) );
+
+		// Register Setting
+		register_setting(
+			'micropub_writing',
+			'micropub_default_post_status', // Setting Name
+			array(
+				'type'              => 'string',
+				'description'       => 'Default Post Status for Micropub Server',
+				'sanitize_callback' => 'sanitize_text_field',
+				'show_in_rest'      => false,
+			)
+		);
+	}
+
+	public static function admin_init() {
+		$cls  = get_called_class();
+		$page = 'micropub';
+		add_settings_section(
+			'micropub_writing',
+			'Micropub Writing Settings',
+			array( 'Micropub_Admin', 'writing_settings' ),
+			$page
+		);
+
+		add_settings_field(
+			'micropub_default_post_status',
+			__( 'Default Status for Micropub Posts', 'micropub' ),
+			array( $cls, 'default_post_status_setting' ),
+			$page,
+			'micropub_writing'
+		);
+	}
+
+	/**
+	 * Add admin menu entry
+	 */
+	public static function admin_menu() {
```

Comment:
this is new, right? in the future, especially for big structural changes like this (or refactorings etc), you generally want to _just_ make the structural change, in its own commit or PR, _without_ functionality changes, and then make functionality changes in a separate commit or PR. makes reviewing easier.